pkg/config/config.go

```diff
@@ -914,9 +914,14 @@ func ResolveSecrets(config Config, origin string) error {
 	return nil
 }
 
-// Avoid log ingestion breaking because of a newline in the API key
-func sanitizeAPIKey(config Config) {
-	config.Set("api_key", strings.TrimSpace(config.GetString("api_key")))
+// SanitizeAPIKey strips newlines and other control characters from a given key.
+func SanitizeAPIKey(config Config, key string) {
+	config.Set(key, Sanitize(config.GetString(key)))
+}
+
+// Sanitize strips newlines and other control characters from a given string.
+func Sanitize(s string) string {
+	return strings.TrimSpace(strings.Split(s, ",")[0])
```

Comment:
I did not catch `strings.Split(s, ",")[0]` during the first review. 
It seems to be very specific to https://github.com/DataDog/datadog-agent/pull/5703/files#diff-21b2309ad06ba5dc719ba11f414fb0d1L419.
What do you think about handling `strings.Split(s, ",")[0]` outside this function (handle it in pkg/process/config/config.go)

Reply:
Right, my bad. It is process specific.